projects/storefrontapp-e2e-cypress/cypress/helpers/customer-ticketing/customer-ticketing.ts

```diff
@@ -0,0 +1,153 @@
+import { clickHamburger } from '../checkout-flow';
+import { waitForPage, visitHomePage } from '../checkout-flow';
+import { loginRegisteredUser as login } from "../cart";
+
+const HTTP_STATUS_OK = 200;
+const FIRST_ROW = 1;
+const SUBJECT_COLUMN = 1;
+const CATEGORY_COLUMN = 2;
+const STATUS_COLUMN = 5;
+
+const CUSTOMER_SUPPORT_MENU_OPTION_INDEX = 14;
+const FIRST_TICKET_COLUMN_INDEX = 1;
+const FIRST_TICKET_ROW_INDEX = 1;
+
+export enum TestCategory {
+  enquiry = "Enquiry",
+  complaint = "Complaint",
+  problem = "Problem",
+}
+
+export interface TestTicketDetails {
+  subject: string;
+  message: string;
+  category: TestCategory;
+  associatedTo?: string;
+  filename?: string;
+}
+
+export function loginRegisteredUser() {
+  login();
+}
+
+export function clickMyAccountMenuOption(){
+  cy.visit('/');
+  cy.get('#cx-header', { timeout: 10000 }).should('be.visible');
+  cy.onMobile(() => {
+    clickHamburger();
+  });
+  cy.get('.accNavComponent button').click();
+}
+
+export function clickCustomerSupportMenuOption(){
+  cy.get(`.accNavComponent li:nth-child(${CUSTOMER_SUPPORT_MENU_OPTION_INDEX})`).should('contain.text', 'Customer Service');
+  cy.get(`.accNavComponent li:nth-child(${CUSTOMER_SUPPORT_MENU_OPTION_INDEX}) a`).click();
+}
+
+export function verifyTicketListingPageVisit(){
+  cy.url().should('include','/my-account/support-tickets');
+  cy.get('cx-customer-ticketing-list').should('exist');
+}
+
+export function clickFirstTicketFromTicketListing(){
+  cy.get(`#ticketing-list-table tbody tr:nth-child(${FIRST_TICKET_ROW_INDEX}) .cx-ticketing-list-data:nth-child(${FIRST_TICKET_COLUMN_INDEX}) a.cx-ticketing-list-value`).should('exist');
+  cy.get(`#ticketing-list-table tbody tr:nth-child(${FIRST_TICKET_ROW_INDEX}) .cx-ticketing-list-data:nth-child(${FIRST_TICKET_COLUMN_INDEX}) a.cx-ticketing-list-value`).click();
+}
+
+export function verifyTicketDetailsPageVisit(){
+  cy.url().should('match',/http:\/\/.+\/my\-account\/support\-ticket\/[0-9]+/);
+  cy.get('cx-customer-ticketing-messages').should('exist');
+}
+
+export function visitElectronicTicketListingPage() {
+  visitHomePage();
+  const supportPage = waitForPage('/my-account/support-tickets', 'support-tickets');
```

Comment:
What does the second parameter means?  "support-tickets"

Reply:
Do we really need waitForPage utility?  Documentation to achieve this is very simple:
```
const ticketListingPageUrl = '/my-account/support-tickets';
cy.intercept(ticketListingPageUrl).as('listTickets');
cy.visit(ticketListingPageUrl);
cy.wait('@listTickets');
```

Reply:
the second parameter is the alias and waitForPage builds the full ticketListingPageUrl (base site + page) inside it.

Reply:
But you are not using the alias, so is it needed? You are using the returned value as an alias.
The direct usage of the interceptor shouldn't be sufficient for this case?

Reply:
The alias is return https://github.com/SAP/spartacus/blob/5aa77f99e780102a71930334d7c23c0ee91475a4/projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts#L81

Reply:
But, that is what we are exactly doing in the code I added
cy.intercept(ticketListingPageUrl).as('listTickets')
equals:
waitForPage(ticketListingPageUrl, 'listTickets');
I don't believe we need the utility function

Reply:
Let's keep the waitForPage since it is more stable than the interceptor